api/src/main/java/org/neo4j/ogm/compiler/CypherEmitter.java

```diff
@@ -15,17 +15,18 @@
 
 import java.util.Map;
 
+import org.neo4j.ogm.utils.Pair;
+
 /**
  * @author Vince Bickers
+ * @author Mark Angrish
  */
 public interface CypherEmitter {
 
     /**
-     * Emits one or more Cypher clauses.
+     * Emits one or more Cypher clauses and their parameters.
      *
-     * @param queryBuilder The {@code StringBuilder} to which the Cypher should be appended
-     * @param parameters A {@link Map} to which Cypher parameter values may optionally be added as the query is built up
+     * @return A Pair tuple containing the Cypher query (first) and parameters (second).
      */
-
-    void emit(StringBuilder queryBuilder, Map<String, Object> parameters);
+    Pair<String, Map<String, Object>> emit();
```

Comment:
Encapsulation is a good idea but I think Pair is a premature abstraction, as we don't use it except in the context of generating cypher statements. Maybe use an explicit class CypherStatement instead?